src/core/layertree/qgslayertreemodel.cpp

```diff
@@ -208,6 +210,7 @@ QVariant QgsLayerTreeModel::data( const QModelIndex &index, int role ) const
 
         case QgsMapLayerType::MeshLayer:
           icon = QgsIconUtils::iconMesh();
+          isModified = qobject_cast<QgsMeshLayer *>( layer )->isFrameModified();
```

Comment:
Would be nicer to make virtual QgsMapLayer::isModified so there's consistent API across layer types